persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stores/AbstractJdbcStore.java

```diff
@@ -0,0 +1,195 @@
+package org.infinispan.persistence.jdbc.stores;
+
+import org.infinispan.commons.io.ByteBuffer;
+import org.infinispan.commons.marshall.StreamingMarshaller;
+import org.infinispan.persistence.jdbc.JdbcUtil;
+import org.infinispan.persistence.jdbc.configuration.AbstractJdbcStoreConfiguration;
+import org.infinispan.persistence.jdbc.connectionfactory.ConnectionFactory;
+import org.infinispan.persistence.jdbc.connectionfactory.ManagedConnectionFactory;
+import org.infinispan.persistence.jdbc.logging.Log;
+import org.infinispan.persistence.jdbc.table.management.TableManager;
+import org.infinispan.persistence.spi.AdvancedLoadWriteStore;
+import org.infinispan.persistence.spi.InitializationContext;
+import org.infinispan.persistence.spi.PersistenceException;
+import org.infinispan.persistence.spi.TransactionalCacheWriter;
+
+import javax.transaction.Transaction;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+/**
+ * @author Ryan Emerson
+ */
+public abstract class AbstractJdbcStore<K,V> implements AdvancedLoadWriteStore<K,V>, TransactionalCacheWriter<K,V> {
+   private final Map<Transaction, Connection> transactionConnectionMap = new ConcurrentHashMap<>();
+   private AbstractJdbcStoreConfiguration configuration;
+   protected ConnectionFactory connectionFactory;
+   protected TableManager tableManager;
+   protected InitializationContext ctx;
+   protected String cacheName;
+   protected Log log;
+
+   protected abstract Log getLog();
```

Comment:
@wburns I have made the change for log, however it is not possible to make this change for ```tableManager``` as it depends on a configuration object which is acquired from the InitializationContext in the init method.